zebra-chain/src/value_balance.rs

```diff
@@ -127,6 +129,33 @@ where
             orchard: zero,
         }
     }
+
+    /// To byte array
+    pub fn to_bytes(self) -> [u8; 32] {
+        let transparent = self.transparent.to_bytes();
+        let sprout = self.sprout.to_bytes();
+        let sapling = self.sapling.to_bytes();
+        let orchard = self.orchard.to_bytes();
+        match [transparent, sprout, sapling, orchard].concat().try_into() {
+            Ok(bytes) => bytes,
+            _ => unreachable!("should be impossible to get here"),
+        }
+    }
+
+    /// From byte array
+    pub fn from_bytes(bytes: [u8; 32]) -> Self {
+        let transparent = Amount::from_bytes(bytes[0..8].try_into().unwrap());
+        let sprout = Amount::from_bytes(bytes[8..16].try_into().unwrap());
+        let sapling = Amount::from_bytes(bytes[16..24].try_into().unwrap());
+        let orchard = Amount::from_bytes(bytes[24..32].try_into().unwrap());
```

Comment:
Same here, i think they are safe but maybe replace with `expect()`?

Reply:
Let's use `expect` with a message that explains why the code is unreachable.